### ISSUE TITLE
Array value assignment and access 

### ISSUE THREAD

**Issue by crusader2000** [cocotb/cocotb#1785]
I am trying to assign value to an input array. 
```
input [7:0] src [0:1023];
```
I tried using this syntax 
```

# Assign a value to a memory deep in the hierarchy
dut.src[0] <= 190
``` 
But I get the following error. 
```
IndexError: src(GPI_REGISTER)  contains no object at index 0
```

How to resolve this?

**Comment by themperek** [comment]
If you want increase chances that someone will know I would suggest to provide:
- cocotb version 
- simulator version
- os
- minimum reproducable example

**Comment by crusader2000** [comment]
Cocotb version -1.3.0
Simulator - iverilog - 10.3
OS - Fedora 29
Python file - https://github.com/crusader2000/prga/blob/dev/cocotb/fifo/prga_fifo_cocotb.py
Top Module - https://github.com/crusader2000/prga/blob/dev/cocotb/fifo/prga_fifo_tb_wrapper.v

**Comment by elgorwi** [comment]
Have a look at the mean example. It has just been updated to support Icarus and does assign values to an array. I recommend pulling the latest from master and reinstalling. Then try running the mean example.

**Comment by ktbarrett** [comment]
This could be resolved in #2079. Try [installing the development version](https://docs.cocotb.org/en/stable/install_devel.html#install-devel) and report back if you are still experiencing this problem.
